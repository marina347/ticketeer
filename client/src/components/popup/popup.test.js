import React from "react";
import { shallow } from "enzyme";
import Popup from "./popup.component";

const Content = () => {
  return <div></div>;
};

describe("Popup component", () => {
  let wrapper;
  let mockOnPopupClose;
  const PopupWithContent = Popup(Content);
  beforeEach(() => {
    mockOnPopupClose = jest.fn();

    wrapper = shallow(<PopupWithContent onPopupClose={mockOnPopupClose} />);
  });

  it("Should render Popup component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("Content should have closePopup prop", () => {
    expect(
      wrapper
        .find('[id="popup"]')
        .find('[id="popup_content"]')
        .find('[id="content"]')
        .prop("closePopup")
    ).not.toEqual(undefined);
  });
});
