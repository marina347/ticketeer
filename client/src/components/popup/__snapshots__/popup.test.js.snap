// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup component Should render Popup component 1`] = `
<div
  className="popup"
  id="popup"
>
  <div
    className="popup__content"
    id="popup_content"
  >
    <FormButton
      class="btn btn-close btn-close--position-2"
      onClick={[MockFunction]}
    >
      Ã—
    </FormButton>
    <Content
      closePopup={[MockFunction]}
      id="content"
    />
  </div>
</div>
`;
